[ADC] Use simple filter for ADC plugin and show value on setup page

TD-er opened the pull request:
When using 'Oversampling' the Analog input plugin is simply summing all samples and upon calling the read function it will divide this sum by the number of samples.
This added filtering is keeping track of the lowest and highest value measured in this loop and excludes these extremes.
This makes the reading a lot more stable, since it is often only a single extreme value which causes noise in the reported values.

Also added a note to the setup page near the calibration points to make it easier to add calibration values. It shows the current raw value + its calibrated equivalent (if enabled).

TD-er committed:
[ADC] Use simple filter for ADC plugin and show value on setup page

When using 'Oversampling' the Analog input plugin is simply summing all samples and upon calling the read function it will divide this sum by the number of samples.
This added filtering is keeping track of the lowest and highest value measured in this loop and excludes these extremes.
This makes the reading a lot more stable, since it is often only a single extreme value which causes noise in the reported values.

Also added a note to the setup page near the calibration points to make it easier to add calibration values. It shows the current raw value + its calibrated equivalent (if enabled).

TD-er committed:
[Travis] Change needed due to change in core 2.6.0 File::write

See   https://github.com/esp8266/Arduino/commit/b1da9eda467cc935307d553692fdde2e670db258#r32622483

TD-er committed:
[Travis] Changes needed due to update core 2.6.0 for File::write

TD-er committed:
[Build] Add build stats to track 

See also [PlatformIO docs](http://docs.platformio.org/en/latest/faq.html#program-memory-usage) and https://github.com/esp8266/Arduino/issues/3740#issuecomment-338416331

TD-er committed:
[Memory] Add several F() macro markers and disable some debug logs

All in order to reduce memory usage.

TD-er committed:
[Modbus] Split Modbus RTU from Senseair plugin

To allow other plugins to use Modbus RTU also.

uzi18 commented on src/WebServer.ino:
```diff
@@ -123,7 +123,7 @@ private:
     if (beforeTXRam < 3000) {
       lowMemorySkip = true;
-      WebServer.send(200, "text/plain", "Low memory. Cannot display webpage :-(");
+      WebServer.send(200, F("text/plain"), "Low memory. Cannot display webpage :-(");
```
WebServer.send(200, F("text/plain"), F("Low memory. Cannot display webpage :-("));

TD-er replied:
You're right :)
This was just a commit to save my work before going to sleep. I've been tracking a lot of these F-macro strings

Edit:
Strange thing is, that it was building fine in Windows.
But it's the other way around, you cannot use F-macros in that class within the Webserver.
F-macros are nice to have but still hard to get it right.

TD-er committed:
[Travis] Build error including F-macro in private class webserver

TD-er committed:
[ADC] Show calibration range + step size

This min/max/stepsize info makes it easier to set the calibration.

TD-er committed:
[Travis] F-macro related build issues

Some core version builds were not using the same disable debug flags.

TD-er committed:
[MPU6050] Reverted to old version before adding Plugin_task_data array

See issue: #2381
Commits:
- https://github.com/letscontrolit/ESPEasy/commit/af20984079d3e7aa59e08fd9b232f6d17ba3b523#diff-ec860ac195fffa61ec11dd419fefa5b9
- https://github.com/letscontrolit/ESPEasy/commit/6400c495e24f39ebac88eb634f29cfb73137fa2b#diff-ec860ac195fffa61ec11dd419fefa5b9

TD-er committed:
[Cleanup] Remove P030 BMP280 since it has become obsolete by BMx280

TD-er committed:
[Webserver] Add documentation link to Advanced settings page

And output the new settings in the info page & JSON output
Not yet included in the JSON used for the new UI.

TD-er committed:
[ADC] Increase readability of ADC calibration settings

Output example:
Current:	405 ≙ 13.034
Minimum:	0 ≙ 1.664
Maximum:	1023 ≙ 30.346
Step size:	1 ≙ 0.028
